roachtest: psycopg failed

roachtest.psycopg [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/21178892?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/21178892?buildTab=artifacts#/psycopg) on release-25.2 @ [e4718ac2068cb3349bf1620f85eb26b66071faff](https://github.com/cockroachdb/cockroach/commits/e4718ac2068cb3349bf1620f85eb26b66071faff):


```
(orm_helpers.go:224).summarizeFailed: 
Tests run on Cockroach v25.2.14-dev-e4718ac2068cb3349bf1620f85eb26b66071faff
Tests run against psycopg 3.2.8
3860 Total Tests Run
3859 tests passed
1 test failed
1895 tests skipped
17 tests ignored
0 tests passed unexpectedly
1 test failed unexpectedly
0 tests expected failed but skipped
0 tests expected failed but not run
---
--- FAIL: tests.pool.test_pool.test_check - unknown (unexpected)
For a full summary look at the psycopg artifacts 
An updated blocklist (psycopgBlockList) is available in the artifacts' psycopg log
test artifacts and logs in: /artifacts/psycopg/cpu_arch=arm64/run_1
```

Parameters:
 - <code>arch=arm64</code>
 - <code>cloud=gce</code>
 - <code>coverageBuild=false</code>
 - <code>cpu=4</code>
 - <code>diskCount=0</code>
 - <code>encrypted=false</code>
 - <code>metamorphicBufferedSender=false</code>
 - <code>metamorphicLeases=default</code>
 - <code>runtimeAssertionsBuild=false</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-21178892/psycopg/1772441034384/1772442214547)

</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*psycopg.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-60919

--- a/pkg/cmd/roachtest/tests/psycopg_blocklist.go
+++ b/pkg/cmd/roachtest/tests/psycopg_blocklist.go
@@ -50,6 +50,7 @@ var psycopgIgnoreList = blocklist{
 	`tests.pool.test_pool_common_async.test_stats_usage[asyncio-AsyncNullConnectionPool]`:            "requires insecure mode",
 	`tests.pool.test_pool_null.test_stats_connect`:                                                   "requires insecure mode",
 	`tests.pool.test_pool_null_async.test_stats_connect[asyncio]`:                                    "requires insecure mode",
+	`tests.pool.test_pool.test_check`:                                                                "flaky; see #164635",
 	`tests.test_connection.test_cancel_safe_error`:                                                   "requires insecure mode",
 	`tests.test_connection_async.test_cancel_safe_error[asyncio]`:                                    "requires insecure mode",
 	`tests.test_generators.test_connect_operationalerror_pgconn`:                                     "requires insecure mode",

